safe-global/safe-wallet-web
# Analytics: track tx type on sign/execute

## What it solves

We'll be now tracking the type of transaction for all tx events (Create/Confirm/Execute).
I've centralized the tracking of these events in the SignOrExecute component.

## How to test it

Create/sign/execute different types of transactions from this list:

```
  // Settings
  owner_add = 'owner_add',
  owner_remove = 'owner_remove',
  owner_swap = 'owner_swap',
  owner_threshold_change = 'owner_threshold_change',

  // Module txs
  guard_remove = 'guard_remove',
  module_remove = 'module_remove',

  // Transfers
  transfer_token = 'transfer_token',
  transfer_nft = 'transfer_nft',

  // Other
  batch = 'batch',
  rejection = 'rejection',
  typed_message = 'typed_message',
  safeapps = 'safeapps',
  walletconnect = 'walletconnect',
  custom = 'custom',
```

It should emit GTM events with the corresponding type in the `eventLabel`.

## Extra changes: merged Safe Open events
@alicecha requested that `Open Safe` and `Sidebar open Safe` events be merged.
Both will now emit `Open Safe` with a label depending on where it happened – `after_create`, `after_add` and `sidebar`.
